db/seeds.rb

```diff
@@ -91,8 +91,14 @@
 
   next if school.placements.any?
 
-  subject = school.phase == "Primary" ? Subject.primary.first : Subject.secondary.first
-  placement = Placement.create!(school:, subject:)
+  year_group = nil
+  subject = if school.phase == "Primary"
+              year_group =  Placement.year_groups.to_a.sample.first
+              Subject.primary.first
+            else
+              Subject.secondary.first
+            end
```

Comment:
It feels a bit weird to be setting variables within the block that sets other variables – it causes a side effect where you'd typically only expect a return value.

Would it be cleaner to structure it like this now, given we have two variables that need to be set?

```suggestion
  if school.phase == "Primary"
    subject = Subject.primary.first
    year_group =  Placement.year_groups.to_a.sample.first
  else
    subject = Subject.secondary.first
    year_group = nil
  end
```

Reply:
resolved in [7fd1169](https://github.com/DFE-Digital/itt-mentor-services/pull/705/commits/7fd1169d7e45ec6e4b52685cfd6c49153565d899)